api/app/modeling/serialize.py

```diff
@@ -0,0 +1,64 @@
+from dataclasses import dataclass
+from pathlib import Path
+import json
+import logging
+import os
+import re
+
+from tensorflow.python.lib.io import file_io
+from transformers import T5Config, T5ForConditionalGeneration
+
+
+__all__ = ['CachedT5ModelLoader']
+TRANSFO_PREFIX = 'model.ckpt'
+
+
+@dataclass
+class CachedT5ModelLoader:
```

Comment:
Can we just use whatever huggingface implements as checkpoint caching?

Reply:
Unfortunately, HF supports neither GCP downloading nor the creation of custom checkpoints. Also, I plan to refactor this module in the future once we have other caching needs.

Reply:
I tried to keep the implementation as close in spirit as possible to HF's:
https://github.com/huggingface/transformers/blob/master/src/transformers/file_utils.py#L70

Reply:
Cool, thanks for the clarification!